### PR TITLE
make sure NC_CHAR attributes returned as ascii strings (issue #529)

### PR DESCRIPTION
**Pull request by jswhit**
In python3, ascii attributes of type NC_CHAR will now be returned as bytes, not strings.  This will potentially break some code.

### CODE REVIEW

File: netCDF4/_netCDF4.pyx [hunk 1]
```diff
@@ -4912,6 +4914,21 @@ cdef _read_enum(group, nc_type xtype, endian=None):
...
+    try:
+        return string.encode(encoding)
+    except UnicodeEncodeError:
+        return string
+
+def _bytes_to_str(b):
+    # if python 3 and b is a bytes object, convert bytes to ascii string.
+    # otherwise do nothing.
+    if python3 and type(b) == bytes:
+        return str(b,encoding='ascii')
```
Comment (jhamman) [inline netCDF4/_netCDF4.pyx]: space after `,`

File: netCDF4/_netCDF4.pyx [hunk 2]
```diff
@@ -5145,6 +5164,7 @@ datetime objects. The datetime instances
 do not contain a time-zone offset, even if the specified `units`
 contains one.
     """
+    units = _bytes_to_str(units); calendar = _bytes_to_str(calendar)
```
Comment (jhamman) [inline netCDF4/_netCDF4.pyx]: I'd put these on separate lines.

**Comment by jhamman** [comment]
@jswhit - two minor comments and one question?  Does this give us complete roundtrip behavior for attributes that are written as python 3 bytestrings  and unicode strings?

**Comment by jswhit** [comment]
With these changes, the attribute types should be preserved on roundtrip, both for python 2 and 3.

**Comment by jhamman** [comment]
Great.  This looks good to me then.

**Comment by jswhit** [comment]
OK, thanks - merging now...

**Pull request merged**